From ba47e6dae95f7f5818d5bccb2720d2513c263f4c Mon Sep 17 00:00:00 2001
From: builder <ksridharan@marvell.com>
Date: Thu, 30 Nov 2023 06:28:36 -0800
Subject: [PATCH] Marvell-teralynx-generate_dump

---
 scripts/generate_dump | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/scripts/generate_dump b/scripts/generate_dump
index 64a89172..e30ee43f 100755
--- a/scripts/generate_dump
+++ b/scripts/generate_dump
@@ -1502,6 +1502,20 @@ collect_broadcom() {
     copy_from_masic_docker "syncd" "/var/log/bcm_diag_post" "/var/log/bcm_diag_post"
 }
 
+##############################################################################
+# collect_innovium
+# Globals:
+#   None
+# Arguments:
+#   None
+# Retuens:
+#   None
+##############################################################################
+collect_innovium() {
+	save_cmd "ivmcmd 'show techsupport -i /innovium/show_techsupport_infile'" "show_techsupport_op_ifcs.log"
+	save_cmd "ivmcmd 'show techsupport -i /innovium/show_techsupport_infile_iSAI'" "show_techsupport_op_iSAI.log"
+}
+
 ###############################################################################
 # Collect Barefoot specific information
 # Globals:
@@ -2033,6 +2047,10 @@ main() {
         collect_broadcom
     fi
 
+    if [ "$asic" = "innovium" ]; then
+        collect_innovium
+    fi
+
     if [ "$asic" = "cisco-8000" ]; then
         collect_cisco_8000
     fi
-- 
2.17.1

