From bfc0ec529b2377bca3a6d06f0f2edf7f3ed560e8 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Thu, 5 Jun 2025 22:04:39 +0300
Subject: [PATCH] [buildimage] marvell-prestera sai version 1.16.1-2

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 platform/marvell-prestera/sai.mk | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/platform/marvell-prestera/sai.mk b/platform/marvell-prestera/sai.mk
index 2191d415e..5cffee2a9 100644
--- a/platform/marvell-prestera/sai.mk
+++ b/platform/marvell-prestera/sai.mk
@@ -1,12 +1,12 @@
 # Marvell SAI
 
-BRANCH = master
+BRANCH = 202505 #touch force to patch
 ifeq ($(CONFIGURED_ARCH),arm64)
-MRVL_SAI_VERSION = 1.15.1-1
+MRVL_SAI_VERSION = 1.16.1-2
 else ifeq ($(CONFIGURED_ARCH),armhf)
-MRVL_SAI_VERSION = 1.15.1-1
+MRVL_SAI_VERSION = 1.16.1-2
 else
-MRVL_SAI_VERSION = 1.15.1-1
+MRVL_SAI_VERSION = 1.16.1-2
 endif
 
 MRVL_SAI_URL_PREFIX = https://github.com/Marvell-switching/sonic-marvell-binaries/raw/master/$(CONFIGURED_ARCH)/sai-plugin/$(BRANCH)/
-- 
2.25.1

