From 5ea7feb8371824fca4a76665e882de00e991dfb5 Mon Sep 17 00:00:00 2001
From: Pavan Naregundi <pnaregundi@marvell.com>
Date: Fri, 26 Jul 2024 09:11:28 +0000
Subject: [PATCH] Update SDK deb version to 1.14.0-1

Signed-off-by: Pavan Naregundi <pnaregundi@marvell.com>
---
 platform/marvell/sai.mk | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/platform/marvell/sai.mk b/platform/marvell/sai.mk
index 3a50da904..257afbb75 100644
--- a/platform/marvell/sai.mk
+++ b/platform/marvell/sai.mk
@@ -1,12 +1,12 @@
 # Marvell SAI
 
-BRANCH = master
+BRANCH = 202405
 ifeq ($(CONFIGURED_ARCH),arm64)
-MRVL_SAI_VERSION = 1.13.0-1
+MRVL_SAI_VERSION = 1.14.0-1
 else ifeq ($(CONFIGURED_ARCH),armhf)
-MRVL_SAI_VERSION = 1.13.0-3
+MRVL_SAI_VERSION = 1.14.0-1
 else
-MRVL_SAI_VERSION = 1.13.0-1
+MRVL_SAI_VERSION = 1.14.0-1
 endif
 
 MRVL_SAI_URL_PREFIX = https://github.com/Marvell-switching/sonic-marvell-binaries/raw/master/$(CONFIGURED_ARCH)/sai-plugin/$(BRANCH)/
-- 
2.25.1

