From a528800ad488d930b5678eb96b2398b4ac3ec3e8 Mon Sep 17 00:00:00 2001
From: Pavan Naregundi <pnaregundi@marvell.com>
Date: Fri, 26 Jul 2024 06:08:01 +0000
Subject: [PATCH] Update mrvl-prestera SDK driver

Signed-off-by: Pavan Naregundi <pnaregundi@marvell.com>
---
 platform/marvell/mrvl-prestera | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/platform/marvell/mrvl-prestera b/platform/marvell/mrvl-prestera
index 5834b7338ff9..36fa3a3f4e31 160000
--- a/platform/marvell/mrvl-prestera
+++ b/platform/marvell/mrvl-prestera
@@ -1 +1 @@
-Subproject commit 5834b7338ff9ac6f03d45ab85568048be1f62199
+Subproject commit 36fa3a3f4e317d8c0c111cc74aafffce12e1546d
