From 167a69758f843671ddd0d8d309e443dc545501dd Mon Sep 17 00:00:00 2001
From: ksridharan <ksridharan@marvell.com>
Date: Tue, 16 Apr 2024 03:05:04 -0700
Subject: [PATCH 7/8]  marvell-teralynx-Use the archive repo for Buster #18678

---
 scripts/build_mirror_config.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/scripts/build_mirror_config.sh b/scripts/build_mirror_config.sh
index bd5312c1b..120640fe1 100755
--- a/scripts/build_mirror_config.sh
+++ b/scripts/build_mirror_config.sh
@@ -21,6 +21,10 @@ if [ "$ARCHITECTURE" == "armhf" ]; then
     DEFAULT_MIRROR_SECURITY_URLS=http://deb.debian.org/debian-security/
 fi
 
+if [ "$DISTRIBUTION" == "buster" ]; then
+    DEFAULT_MIRROR_URLS=http://archive.debian.org/debian/
+fi
+
 if [ "$MIRROR_SNAPSHOT" == y ]; then
     if [ -f "$MIRROR_VERSION_FILE" ]; then
         DEBIAN_TIMESTAMP=$(grep "^debian==" $MIRROR_VERSION_FILE | tail -n 1 | sed 's/.*==//')
-- 
2.17.1

