From f3e499f2a98a823b3f392397a2ce08cef7a0cac7 Mon Sep 17 00:00:00 2001
From: ksridharan <ksridharan@marvell.com>
Date: Fri, 19 Apr 2024 20:45:19 -0700
Subject: [PATCH 8/8] marvell-teralynx-Add SDK dependent python packages #18232

---
 platform/innovium/docker-syncd-invm/Dockerfile.j2 | 1 +
 1 file changed, 1 insertion(+)

diff --git a/platform/innovium/docker-syncd-invm/Dockerfile.j2 b/platform/innovium/docker-syncd-invm/Dockerfile.j2
index aa608b1de..f477c3d9b 100755
--- a/platform/innovium/docker-syncd-invm/Dockerfile.j2
+++ b/platform/innovium/docker-syncd-invm/Dockerfile.j2
@@ -20,6 +20,7 @@ RUN apt-get install -y libjansson4
 RUN apt-get install -y libyaml-dev
 RUN apt-get install -y binutils
 RUN pip3 install numpy
+RUN pip3 install yamlordereddictloader
 
 RUN dpkg -i \
 {% for deb in docker_syncd_invm_debs.split(' ') -%}
-- 
2.17.1

