From 7b44f2cddaec287f33fc6b2513c848c074b86543 Mon Sep 17 00:00:00 2001
From: apannerselva <apannerselva@marvell.com>
Date: Sat, 4 Apr 2026 06:58:58 +0530
Subject: [PATCH] FRR Series list

Signed-off-by: apannerselva <apannerselva@marvell.com>
---
 src/sonic-frr/patch/series | 61 +++++++++++++++++++++++++++++++++++---
 1 file changed, 57 insertions(+), 4 deletions(-)

diff --git a/src/sonic-frr/patch/series b/src/sonic-frr/patch/series
index cde37b0ae..ed87ce2e8 100644
--- a/src/sonic-frr/patch/series
+++ b/src/sonic-frr/patch/series
@@ -15,9 +15,63 @@
 0016-SONiC-ONLY-Set-multipath-to-514-and-disable-bgp-vnc-for-optimiz.patch
 0017-SONiC-ONLY-Patch-to-send-tag-value-associated-with-route-via-ne.patch
 0018-SONiC-ONLY-SRv6-vpn-route-and-sidlist-install.patch
-0022-SONiC-ONLY-This-error-happens-when-we-try-to-write-to-a-socket.patch
-0026-SONiC-ONLY-Translate-tableid-for-dplane-route-notify.patch
-0027-Dont-skip-kernel-routes-uninstall.patch
+0019-SONiC-ONLY-This-error-happens-when-we-try-to-write-to-a-socket.patch
+0020-SONiC-ONLY-Translate-tableid-for-dplane-route-notify.patch
+0021-Dont-skip-kernel-routes-uninstall.patch
+0022-bgpd-Prevent-unnecessary-re-install-of-routes.patch
+0023-zebra-EVPN-VXLAN-Multihome-extern-mode-Enable-kernel.patch
+0024-zebra-New-protocol-RTPROT_HW-for-hardware-learnt-MAC.patch
+0025-doc-zebra-Doc-update-for-zebra-option-kernel-mac-ext.patch
+0026-zebra-EVPN-VXLAN-MH-extern-mode-Update-debug-print-f.patch
+0027-zebra-ARP-ND-Suppression-disabled-for-kernel-mac-ext.patch
+0028-zebra-netlink-message-handling-for-mac_extern_learn-.patch
+0029-zebra-mac-sync-update-del-expiry-in-extern_learn-mod.patch
+0030-tests-topotests-EVPN-VXLAN-MH-extern-learn-mode-test.patch
+0031-bgpd-install-evpn-failure-logging.patch
+0032-lib-zebra-Make-Zebra-IPv6-VTEP-Aware.patch
+0033-zebra-Use-RTA_VIA-for-IPv4-hosts-over-IPv6-overlay-f.patch
+0034-bgpd-Make-BGP-V6-VTEP-Aware-for-Single-Homing-VxLAN.patch
+0035-bgpd-zebra-tests-Enable-V6-VTEP-for-Single-Homing-Vx.patch
+0036-tests-Add-V6-VTEP-variant-of-bgp_evpn_vxlan_topo1.patch
+0037-tests-Add-V6-VTEP-variant-for-bgp_evpn_vxlan_macvrf_.patch
+0038-doc-Extend-BGP-and-EVPN-to-reference-IPv6-VTEP-Suppo.patch
+0039-bgpd-EVPNv6-MH-es-vtep-add-del-v6-aware.patch
+0040-zebra-EVPNv6-MH-remote-es-parse-ipaddr.patch
+0041-bgpd-zebra-EVPNv6-MH-esi-v6-aware.patch
+0042-bgpd-EVPNv6-check-vni-routes-nh-attr-v6-aware.patch
+0043-bgpd-EVPNv6-path-nh-link-check-v4-vpn4-for-len.patch
+0044-lib-EVPNv6-MH-increase-pfx-string-len-for-v6-address.patch
+0045-bgpd-EVPNv6-MH-type-1-vni-copy-v6-aware.patch
+0046-bgpd-EVPNv6-MH-type-1-global-copy-v6-aware.patch
+0047-bgpd-EVPNv6-MH-es-show-cmd-display-v6-vtep.patch
+0048-lib-EVPNv6-MH-fix-type-1-ead-print.patch
+0049-zebra-EVPN-remote-es-debug-to-add-df-alg-param.patch
+0050-bgpd-EVPNv6-MH-debug-print-v4-v6-addr.patch
+0051-tests-EVPNv6-MH-v6-vtep-aware-BGP-num-test.patch
+0052-tests-EVPNv6-MH-load-torm11-cfg-using-frr-reload.patch
+0053-tests-EVPNv6-MH-use-linux-cmds-for-intf-addr.patch
+0054-tests-EVPNv6-MH-v6-vtep-aware-check-ip-addr-after-cf.patch
+0055-tests-EVPNv6-MH-v6-vtep-aware-svd-multiple-vnis.patch
+0056-tests-EVPNv6-MH-v6-vtep-aware-unified-vxlan-cfg.patch
+0057-tests-EVPNv6-MH-v6-vtep-aware-tenant-vrf-ips.patch
+0058-tests-EVPNv6-MH-v6-vtep-aware-topology-doc-formattin.patch
+0059-tests-EVPNv6-mh-v6-vtep-aware-enable-pim6d.patch
+0060-doc-EVPNv6-MH-capture-v6-support.patch
+0061-bgpd-EVPNv6-MH-lttng-traces-IPv6-vtep-aware.patch
+0062-bgpd-EVPNv6-MH-sync-v6-vtep-nhg.patch
+0063-bgpd-EVPNv6-MH-utility-api-to-parse-pi-to-vtep.patch
+0064-bgpd-EVPNv6-MH-utility-to-get-vtep-from-path-attr.patch
+0065-bgpd-EVPNv6-MH-utility-api-vtep-ip-to-attr-nh.patch
+0066-bgpd-EVPNv6-MH-init-using-macro.patch
+0067-bgpd-EVPNv6-MH-init-type4-prefix.patch
+0068-Trigger-neighbour-probe-when-the-hold-timer-starts.patch
+0069-SVI-Deletion-handling-BGP-sync-message-145-159.patch
+0070-zebra-Remove-restriction-to-reachable-neighbors-in-E.patch
+0071-include-linux-Update-NTF_E_MH_PEER_SYNC-flag-value-i.patch
+0072-zebra-Add-FPM-EVPN-MH-message-types-and-structures.patch
+0073-zebra-Add-vid-and-BR_PORT_DELETE-support-for-EVPN-MH.patch
+0062-ospf6d-Add-automatic-MTU-change-detection-for-FULL-a.patch
+0084-support-vrrp6-commands-and-tracking-interface.patch
 0102-staticd-Fix-SRv6-SID-use-after-free-on-locator-deletion.patch
 0103-SONiC-ONLY-bgpd-reduce-suppress-fib-advertisement-delay-to-50ms.patch
 0104-SONiC-ONLY-zebra-skip-if-add-update-in-speed-timer-for-unready-ifp.patch
@@ -27,4 +81,3 @@
 0103-ospf6d-Fix-stack-buffer-overflow-in-Link-LSA-origina.patch
 0104-bgpd-Set-strict-mode-flag-when-we-configure-BFD-hold.patch
 0105-bgpd-Start-BFD-hold-timer-ONLY-if-hold-time-is-confi.patch
-evpn-mh-frr-zebra.patch
-- 
2.25.1

