From 258681e214f56d4f090179ce5951ec84fa560aea Mon Sep 17 00:00:00 2001
From: apannerselva <apannerselva@marvell.com>
Date: Wed, 5 Nov 2025 12:02:51 +0530
Subject: [PATCH] ARS SWSS COMMON changes

Signed-off-by: apannerselva <apannerselva@marvell.com>
---
 common/schema.h | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/common/schema.h b/common/schema.h
index 92703d1..251e3ac 100644
--- a/common/schema.h
+++ b/common/schema.h
@@ -349,6 +349,16 @@ after libswsscommon deb make.
 #define CFG_WATERMARK_TABLE_NAME "WATERMARK_TABLE"
 #define CFG_PFC_PRIORITY_TO_QUEUE_MAP_TABLE_NAME    "MAP_PFC_PRIORITY_TO_QUEUE"
 #define CFG_TC_TO_DOT1P_MAP_TABLE_NAME              "TC_TO_DOT1P_MAP"
+
+#define CFG_ARS_PROFILE                             "ARS_PROFILE"
+#define CFG_ARS_INTERFACE                           "ARS_INTERFACE"
+#define CFG_ARS_OBJECT                              "ARS_OBJECT"
+#define CFG_ARS_NEXTHOP                             "ARS_NEXTHOP"
+
+#define CFG_BUFFER_POOL_TABLE_NAME                  "BUFFER_POOL"
+#define CFG_BUFFER_PROFILE_TABLE_NAME               "BUFFER_PROFILE"
+#define CFG_BUFFER_QUEUE_TABLE_NAME                 "BUFFER_QUEUE"
+#define CFG_BUFFER_PG_TABLE_NAME                    "BUFFER_PG"
 #define CFG_BUFFER_PORT_INGRESS_PROFILE_LIST_NAME   "BUFFER_PORT_INGRESS_PROFILE_LIST"
 #define CFG_BUFFER_PORT_EGRESS_PROFILE_LIST_NAME    "BUFFER_PORT_EGRESS_PROFILE_LIST"
 #define CFG_DEFAULT_LOSSLESS_BUFFER_PARAMETER       "DEFAULT_LOSSLESS_BUFFER_PARAMETER"
@@ -545,6 +555,11 @@ after libswsscommon deb make.
 #define BMP_STATE_BGP_RIB_IN_TABLE               "BGP_RIB_IN_TABLE"
 #define BMP_STATE_BGP_RIB_OUT_TABLE              "BGP_RIB_OUT_TABLE"
 
+/* ARS */
+#define STATE_ARS_PROFILE_TABLE_NAME            "ARS_PROFILE_TABLE"
+#define STATE_ARS_CAPABILITY_TABLE_NAME         "ARS_CAPABILITY_TABLE"
+
+
 #ifdef __cplusplus
 }
 #endif
-- 
2.25.1

