diff --git a/orchagent/Makefile.am b/orchagent/Makefile.am
index f431713..63f65b6 100644
--- a/orchagent/Makefile.am
+++ b/orchagent/Makefile.am
@@ -9,6 +9,7 @@ dist_swss_DATA = \
 		 pfc_detect_innovium.lua  \
 		 pfc_detect_mellanox.lua  \
 		 pfc_detect_broadcom.lua \
+		 pfc_detect_marvell.lua \
 		 pfc_detect_barefoot.lua \
 		 pfc_detect_nephos.lua \
 		 pfc_restore.lua \
diff --git a/orchagent/orchdaemon.cpp b/orchagent/orchdaemon.cpp
index 6bc362f..0006d0b 100644
--- a/orchagent/orchdaemon.cpp
+++ b/orchagent/orchdaemon.cpp
@@ -14,7 +14,7 @@ using namespace std;
 using namespace swss;
 
 /* select() function timeout retry time */
-#define SELECT_TIMEOUT 1000
+#define SELECT_TIMEOUT 1999999
 #define PFC_WD_POLL_MSECS 100
 
 extern sai_switch_api_t*           sai_switch_api;
@@ -428,6 +428,7 @@ bool OrchDaemon::init()
     }
     else if ((platform == INVM_PLATFORM_SUBSTRING)
              || (platform == BFN_PLATFORM_SUBSTRING)
+             || (platform == MRVL_PLATFORM_SUBSTRING)
              || (platform == NPS_PLATFORM_SUBSTRING))
     {
 
@@ -459,7 +460,7 @@ bool OrchDaemon::init()
 
         static const vector<sai_queue_attr_t> queueAttrIds;
 
-        if ((platform == INVM_PLATFORM_SUBSTRING) || (platform == NPS_PLATFORM_SUBSTRING))
+        if ((platform == INVM_PLATFORM_SUBSTRING) || (platform == NPS_PLATFORM_SUBSTRING) || (platform == MRVL_PLATFORM_SUBSTRING))
         {
             m_orchList.push_back(new PfcWdSwOrch<PfcWdZeroBufferHandler, PfcWdLossyHandler>(
                         m_configDb,
diff --git a/orchagent/pfc_detect_marvell.lua b/orchagent/pfc_detect_marvell.lua
index e69de29..6e5b710 100644
--- a/orchagent/pfc_detect_marvell.lua
+++ b/orchagent/pfc_detect_marvell.lua
@@ -0,0 +1,99 @@
+-- KEYS - queue IDs
+-- ARGV[1] - counters db index
+-- ARGV[2] - counters table name
+-- ARGV[3] - poll time interval (milliseconds)
+-- return queue Ids that satisfy criteria
+
+local counters_db = ARGV[1]
+local counters_table_name = ARGV[2]
+local poll_time = tonumber(ARGV[3]) * 1000
+
+local rets = {}
+
+redis.call('SELECT', counters_db)
+
+-- Iterate through each queue
+local n = table.getn(KEYS)
+for i = n, 1, -1 do
+    local counter_keys = redis.call('HKEYS', counters_table_name .. ':' .. KEYS[i])
+    local counter_num = 0
+    local old_counter_num = 0
+    local is_deadlock = false
+    local pfc_wd_status = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'PFC_WD_STATUS')
+    local pfc_wd_action = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'PFC_WD_ACTION')
+    local big_red_switch_mode = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'BIG_RED_SWITCH_MODE')
+    if not big_red_switch_mode and (pfc_wd_status == 'operational' or pfc_wd_action == 'alert') then
+        local detection_time = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'PFC_WD_DETECTION_TIME')
+        if detection_time then
+            detection_time = tonumber(detection_time)
+            local time_left = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'PFC_WD_DETECTION_TIME_LEFT')
+            if not time_left  then
+                time_left = detection_time
+            else
+                time_left = tonumber(time_left)
+            end
+
+            local queue_index = redis.call('HGET', 'COUNTERS_QUEUE_INDEX_MAP', KEYS[i])
+            local port_id = redis.call('HGET', 'COUNTERS_QUEUE_PORT_MAP', KEYS[i])
+            local pfc_rx_pkt_key = 'SAI_PORT_STAT_PFC_' .. queue_index .. '_RX_PKTS'
+            local pfc_duration_key = 'SAI_PORT_STAT_PFC_' .. queue_index .. '_RX_PAUSE_DURATION'
+
+            -- Get all counters
+            local occupancy_bytes = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'SAI_QUEUE_STAT_CURR_OCCUPANCY_BYTES')
+            local packets = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'SAI_QUEUE_STAT_PACKETS')
+            local pfc_rx_packets = redis.call('HGET', counters_table_name .. ':' .. port_id, pfc_rx_pkt_key)
+            local pfc_duration = "0"
+
+            if occupancy_bytes and packets and pfc_rx_packets and pfc_duration then
+                occupancy_bytes = tonumber(occupancy_bytes)
+                packets = tonumber(packets)
+                pfc_rx_packets = tonumber(pfc_rx_packets)
+                pfc_duration =  tonumber(pfc_duration)
+
+                local packets_last = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'SAI_QUEUE_STAT_PACKETS_last')
+                local pfc_rx_packets_last = redis.call('HGET', counters_table_name .. ':' .. port_id, pfc_rx_pkt_key .. '_last')
+                local pfc_duration_last = redis.call('HGET', counters_table_name .. ':' .. port_id, pfc_duration_key .. '_last')
+                -- DEBUG CODE START. Uncomment to enable
+                local debug_storm = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'DEBUG_STORM')
+                -- DEBUG CODE END.
+
+                -- If this is not a first run, then we have last values available
+                if packets_last and pfc_rx_packets_last and pfc_duration_last then
+                    packets_last = tonumber(packets_last)
+                    pfc_rx_packets_last = tonumber(pfc_rx_packets_last)
+                    pfc_duration_last = tonumber(pfc_duration_last)
+
+                    -- Check actual condition of queue being in PFC storm
+                    if (occupancy_bytes > 0 and packets - packets_last == 0 and pfc_rx_packets - pfc_rx_packets_last > 0) or
+                        -- DEBUG CODE START. Uncomment to enable
+                        (debug_storm == "enabled") or
+                        -- DEBUG CODE END.
+                        (occupancy_bytes == 0 and packets - packets_last == 0 and (pfc_duration - pfc_duration_last) > poll_time * 0.8) then
+                        if time_left <= poll_time then
+                            redis.call('PUBLISH', 'PFC_WD_ACTION', '["' .. KEYS[i] .. '","storm"]')
+                            is_deadlock = true
+                            time_left = detection_time
+                        else
+                            time_left = time_left - poll_time
+                        end
+                    else
+                        if pfc_wd_action == 'alert' and pfc_wd_status ~= 'operational' then
+                            redis.call('PUBLISH', 'PFC_WD_ACTION', '["' .. KEYS[i] .. '","restore"]')
+                        end
+                        time_left = detection_time
+                    end
+                end
+
+            -- Save values for next run
+                redis.call('HSET', counters_table_name .. ':' .. KEYS[i], 'SAI_QUEUE_STAT_PACKETS_last', packets)
+                redis.call('HSET', counters_table_name .. ':' .. KEYS[i], 'PFC_WD_DETECTION_TIME_LEFT', time_left)
+                redis.call('HSET', counters_table_name .. ':' .. port_id, pfc_rx_pkt_key .. '_last', pfc_rx_packets)
+                redis.call('HDEL', counters_table_name .. ':' .. port_id, pfc_duration_key .. '_last')
+                redis.call('HSET', counters_table_name .. ':' .. port_id, pfc_duration_key .. '_last', pfc_duration)
+            end
+        end
+    end
+end
+
+return rets
+
diff --git a/orchagent/switchorch.cpp b/orchagent/switchorch.cpp
index 77c8336..c393e32 100644
--- a/orchagent/switchorch.cpp
+++ b/orchagent/switchorch.cpp
@@ -23,7 +23,8 @@ const map<string, sai_switch_attr_t> switch_attribute_map =
     {"lag_hash_seed",                       SAI_SWITCH_ATTR_LAG_DEFAULT_HASH_SEED},
     {"fdb_aging_time",                      SAI_SWITCH_ATTR_FDB_AGING_TIME},
     {"vxlan_port",                          SAI_SWITCH_ATTR_VXLAN_DEFAULT_PORT},
-    {"vxlan_router_mac",                    SAI_SWITCH_ATTR_VXLAN_DEFAULT_ROUTER_MAC}
+    {"vxlan_router_mac",                    SAI_SWITCH_ATTR_VXLAN_DEFAULT_ROUTER_MAC},
+    {"sdk_diag_shell",                      SAI_SWITCH_ATTR_SWITCH_SHELL_ENABLE}
 };
 
 const map<string, sai_packet_action_t> packet_action_map =
@@ -191,6 +192,10 @@ void SwitchOrch::doAppSwitchTableTask(Consumer &consumer)
                         memcpy(attr.value.mac, mac_addr.getMac(), sizeof(sai_mac_t));
                         break;
 
+                   case SAI_SWITCH_ATTR_SWITCH_SHELL_ENABLE:
+                        attr.value.booldata = to_uint<uint32_t>(value);
+                        break;
+
                     default:
                         invalid_attr = true;
                         break;
