diff --git a/debian/control b/debian/control
index 57ce1d7..5945238 100644
--- a/debian/control
+++ b/debian/control
@@ -2,12 +2,12 @@ Source: ptf
 Maintainer: Guohan Lu <lguohan@gmail.com>
 Section: python
 Priority: optional
-Build-Depends: python-all (>= 2.6.6-3), debhelper (>= 9)
+Build-Depends: python3-all (>= 2.6.6-3), debhelper (>= 9)
 Standards-Version: 3.9.1
 
 Package: python-ptf
 Architecture: all
-Depends: ${misc:Depends}, ${python:Depends}, python-scapy
+Depends: ${misc:Depends}, ${python:Depends}, python3-scapy
 Description: PTF is a Python based dataplane test framework.
  .
  PTF is a Python based dataplane test framework. It is based on unittest, which is included in the standard Python distribution.
diff --git a/debian/rules b/debian/rules
index a6dd278..69f38c0 100755
--- a/debian/rules
+++ b/debian/rules
@@ -4,7 +4,7 @@
 # Mon, 19 Oct 2015 12:49:43 -0700
 export PYBUILD_NAME=ptf
 %:
-	dh $@ --with python2 --buildsystem=pybuild
+	dh $@ --with python3 --buildsystem=pybuild
 
 override_dh_auto_test:
 
