From d16ed85dc8aa132ca22978784b2cd388de8c37a3 Mon Sep 17 00:00:00 2001
From: Pavan Naregundi <pnaregundi@marvell.com>
Date: Mon, 20 Jan 2025 16:24:05 +0530
Subject: [PATCH 1/2] [Marvell] Update sai debian

Signed-off-by: Pavan Naregundi <pnaregundi@marvell.com>
---
 platform/marvell/sai.mk | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/platform/marvell/sai.mk b/platform/marvell/sai.mk
index 87bd92448d5d..2191d415e28a 100644
--- a/platform/marvell/sai.mk
+++ b/platform/marvell/sai.mk
@@ -2,11 +2,11 @@
 
 BRANCH = master
 ifeq ($(CONFIGURED_ARCH),arm64)
-MRVL_SAI_VERSION = 1.14.0-1
+MRVL_SAI_VERSION = 1.15.1-1
 else ifeq ($(CONFIGURED_ARCH),armhf)
-MRVL_SAI_VERSION = 1.14.0-1
+MRVL_SAI_VERSION = 1.15.1-1
 else
-MRVL_SAI_VERSION = 1.14.0-1
+MRVL_SAI_VERSION = 1.15.1-1
 endif
 
 MRVL_SAI_URL_PREFIX = https://github.com/Marvell-switching/sonic-marvell-binaries/raw/master/$(CONFIGURED_ARCH)/sai-plugin/$(BRANCH)/

From d37e54c8ebb2684f1d12b17e96ece01db00c6309 Mon Sep 17 00:00:00 2001
From: Pavan Naregundi <pnaregundi@marvell.com>
Date: Mon, 20 Jan 2025 16:27:56 +0530
Subject: [PATCH 2/2] [Marvell] Update SDK driver

Signed-off-by: Pavan Naregundi <pnaregundi@marvell.com>
---
 platform/marvell/mrvl-prestera | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/platform/marvell/mrvl-prestera b/platform/marvell/mrvl-prestera
index f3735bec489c..88f0532a0e6c 160000
--- a/platform/marvell/mrvl-prestera
+++ b/platform/marvell/mrvl-prestera
@@ -1 +1 @@
-Subproject commit f3735bec489c8475d2dcc0b0b5d5a810a6ee919e
+Subproject commit 88f0532a0e6c33d399dc69096da463ce4fcfa8d4
