diff --git a/scripts/portstat b/scripts/portstat
index 42757d3..81761ea 100644
--- a/scripts/portstat
+++ b/scripts/portstat
@@ -251,13 +251,13 @@ class Portstat(object):
                     table.append((key, self.get_port_state(key),
                                   ns_diff(cntr.rx_ok, old_cntr.rx_ok),
                                   ns_brate(cntr.rx_byt, old_cntr.rx_byt, time_gap),
-                                  ns_util(cntr.rx_byt, old_cntr.rx_byt, time_gap),
+                                  ns_util(cntr.rx_byt, old_cntr.rx_byt, time_gap, port_speed),
                                   ns_diff(cntr.rx_err, old_cntr.rx_err),
                                   ns_diff(cntr.rx_drop, old_cntr.rx_drop),
                                   ns_diff(cntr.rx_ovr, old_cntr.rx_ovr),
                                   ns_diff(cntr.tx_ok, old_cntr.tx_ok),
                                   ns_brate(cntr.tx_byt, old_cntr.tx_byt, time_gap),
-                                  ns_util(cntr.tx_byt, old_cntr.tx_byt, time_gap),
+                                  ns_util(cntr.tx_byt, old_cntr.tx_byt, time_gap, port_speed),
                                   ns_diff(cntr.tx_err, old_cntr.tx_err),
                                   ns_diff(cntr.tx_drop, old_cntr.tx_drop),
                                   ns_diff(cntr.tx_ovr, old_cntr.tx_ovr)))
diff --git a/utilities_common/netstat.py b/utilities_common/netstat.py
index 9bf266e..a8d62b5 100755
--- a/utilities_common/netstat.py
+++ b/utilities_common/netstat.py
@@ -23,10 +23,10 @@ def ns_brate(newstr, oldstr, delta):
         return STATUS_NA
     else:
         rate = int(ns_diff(newstr, oldstr).replace(',',''))/delta
-        if rate > 1024*1024*10:
-            rate = "{:.2f}".format(rate/1024/1024)+' MB'
-        elif rate > 1024*10:
-            rate = "{:.2f}".format(rate/1024)+' KB'
+        if rate > 1000*1000*10:
+            rate = "{:.2f}".format(rate/1000/1000)+' MB'
+        elif rate > 1000*10:
+            rate = "{:.2f}".format(rate/1000)+' KB'
         else:
             rate = "{:.2f}".format(rate)+' B'
         return rate+'/s'
@@ -49,7 +49,7 @@ def ns_util(newstr, oldstr, delta, port_rate=PORT_RATE):
         return STATUS_NA
     else:
         rate = int(ns_diff(newstr, oldstr).replace(',',''))/delta
-        util = rate/(port_rate*1024*1024*1024/8.0)*100
+        util = rate/(port_rate*1000*1000*1000/8.0)*100
         return "{:.2f}%".format(util)
 
 def table_as_json(table, header):
