diff --git a/sonic_installer/main.py b/sonic_installer/main.py
index 8696c65..ccfd869 100644
--- a/sonic_installer/main.py
+++ b/sonic_installer/main.py
@@ -413,7 +413,10 @@ def install(url, force, skip_migration=False):
         if skip_migration:
             click.echo("Skipping configuration migration as requested in the command option.")
         else:
-            run_command('config-setup backup')
+            run_command('echo config-setup backup')
+        run_command("rm -rf /host/old_config")
+        # copy directories and preserve original file structure, attributes and associated metadata
+        run_command("cp -ar /etc/sonic /host/old_config")
 
     # Finally, sync filesystem
     run_command("sync;sync;sync")
