diff --git a/files/image_config/syslog/host_umount.sh b/files/image_config/syslog/host_umount.sh
index cedbb5b..285d3de 100755
--- a/files/image_config/syslog/host_umount.sh
+++ b/files/image_config/syslog/host_umount.sh
@@ -5,7 +5,8 @@
 
 journal_stop() {
     systemctl stop systemd-journald.service
-    systemctl stop systemd-journald.socket
+    # socket starts listening during reboot stop and blocks shutdown
+    # systemctl stop systemd-journald.socket
     systemctl stop systemd-journald-audit.socket
     systemctl stop systemd-journald-dev-log.socket
 }
