diff --git a/config/main.py b/config/main.py
index 3b92f66..1876acd 100755
--- a/config/main.py
+++ b/config/main.py
@@ -9,6 +9,7 @@ import re
 import syslog
 import time
 import netifaces
+import json
 
 import sonic_device_util
 import ipaddress
@@ -3312,5 +3313,44 @@ def autorestart(container_name, autorestart_status):
 
     config_db.mod_entry('CONTAINER_FEATURE', container_name, {'auto_restart': autorestart_status})
 
+@config.group(cls=AbbreviationGroup)
+@click.pass_context
+def switch(ctx):
+    """switch related config"""
+    pass
+
+@switch.command('sdk_shell')
+@click.pass_context
+@click.argument('en', metavar='<en>', required=True, type=click.Choice(['enable', 'disable']))
+def set_sdk_diag_shell(ctx, en):
+    import json
+    tmp_file = "/tmp/swss.json"
+    json_file = "/etc/swss/config.d/switch.json"
+
+    try:
+        os.remove(tmp_file)
+    except:
+        pass
+
+    command = " docker cp   swss:" + json_file + " " + tmp_file
+    print(command)
+    subprocess.Popen(command, shell=True)
+    time.sleep(2)
+
+    with open (tmp_file, "r") as f:
+        cfg = json.load(f)
+
+    cfg[0]["SWITCH_TABLE:switch"]["sdk_diag_shell"] = 1 if en == 'enable' else 0
+
+    with open (tmp_file, "w") as f:
+        json.dump(cfg,f,indent=4, ensure_ascii=False)
+
+    command = " docker cp  " + tmp_file +" "+ "swss:"+json_file
+    subprocess.Popen(command, shell=True)
+    time.sleep(2)
+
+    command = "docker exec swss swssconfig " + json_file
+    subprocess.Popen(command, shell=True)
+
 if __name__ == '__main__':
     config()
