diff --git a/scripts/generate_dump b/scripts/generate_dump
index 01e45f8..2e5abc6 100755
--- a/scripts/generate_dump
+++ b/scripts/generate_dump
@@ -182,7 +182,7 @@ save_nat_info() {
 save_proc() {
     local procfiles="$@"
     $MKDIR $V -p $TARDIR/proc \
-        && $CP $V -r $procfiles $TARDIR/proc \
+        && (for f in $procfiles; do ( [ -e $f ] && $CP $V -r $f $TARDIR/proc ) || echo "$f not found" > $TARDIR/$f; done) \
         && $TAR $V -rhf $TARFILE -C $DUMPDIR --mode=+rw $BASE/proc \
         && $RM $V -rf $TARDIR/proc
 }
@@ -461,7 +461,7 @@ main() {
     $MKDIR $V -p $LOGDIR
     $LN $V -s /etc $TARDIR/etc
 
-    ($TAR $V -rhf $TARFILE -C $DUMPDIR --mode=+rw \
+    ($TAR $V -rf $TARFILE -C $DUMPDIR --mode=+rw \
         --exclude="etc/alternatives" \
         --exclude="*/etc/passwd*" \
         --exclude="*/etc/shadow*" \
@@ -506,7 +506,7 @@ main() {
     done
 
     # archive kernel dump files
-    for file in $(find_files "/var/crash/"); do
+    [ -e /var/crash/ ] && for file in $(find_files "/var/crash/"); do
         # don't gzip already-gzipped dmesg files :)
         if [ ! ${file} = "/var/crash/kexec_cmd" -a ! ${file} = "/var/crash/export" ]; then
             if [[ ${file} == *"kdump."* ]]; then
@@ -517,10 +517,12 @@ main() {
         fi
     done
 
+    if [[ "$asic" = "mellanox" ]]; then
     # run 'hw-management-generate-dump.sh' script and save the result file
     /usr/bin/hw-management-generate-dump.sh
     save_file "/tmp/hw-mgmt-dump*" "hw-mgmt" false
     rm -f /tmp/hw-mgmt-dump*
+    fi
 
     # clean up working tar dir before compressing
     $RM $V -rf $TARDIR
