From 9ca11153b3a4e82043f87ab80cb2fc564781fc54 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Thu, 5 Jun 2025 22:04:39 +0300
Subject: [PATCH] [buildimage] marvell-prestera sai version 1.16.1-1

Signed-off-by: Yan Markman <ymarkman@marvell.com>

diff --git a/platform/marvell-prestera/sai.mk b/platform/marvell-prestera/sai.mk
index 2191d415e..0ecbda7f6 100644
--- a/platform/marvell-prestera/sai.mk
+++ b/platform/marvell-prestera/sai.mk
@@ -2,11 +2,11 @@
 
 BRANCH = master
 ifeq ($(CONFIGURED_ARCH),arm64)
-MRVL_SAI_VERSION = 1.15.1-1
+MRVL_SAI_VERSION = 1.16.1-1
 else ifeq ($(CONFIGURED_ARCH),armhf)
-MRVL_SAI_VERSION = 1.15.1-1
+MRVL_SAI_VERSION = 1.16.1-1
 else
-MRVL_SAI_VERSION = 1.15.1-1
+MRVL_SAI_VERSION = 1.16.1-1
 endif
 
 MRVL_SAI_URL_PREFIX = https://github.com/Marvell-switching/sonic-marvell-binaries/raw/master/$(CONFIGURED_ARCH)/sai-plugin/$(BRANCH)/
-- 
2.25.1

