From 7d1aabc1a1ee65cc7316b9781e5024b2dbe9bd70 Mon Sep 17 00:00:00 2001
From: Rajkumar P R <rpennadamram@marvell.com>
Date: Fri, 18 Oct 2024 06:35:06 +0000
Subject: [PATCH] [Marvell] Update amd64 SAI deb version to 1.14.0-2

Signed-off-by: Rajkumar P R <rpennadamram@marvell.com>
---
 platform/marvell/sai.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/platform/marvell/sai.mk b/platform/marvell/sai.mk
index 44c1f92e8..c9fcc1724 100644
--- a/platform/marvell/sai.mk
+++ b/platform/marvell/sai.mk
@@ -6,7 +6,7 @@ MRVL_SAI_VERSION = 1.14.0-2
 else ifeq ($(CONFIGURED_ARCH),armhf)
 MRVL_SAI_VERSION = 1.14.0-1
 else
-MRVL_SAI_VERSION = 1.14.0-1
+MRVL_SAI_VERSION = 1.14.0-2
 endif
 
 MRVL_SAI_URL_PREFIX = https://github.com/Marvell-switching/sonic-marvell-binaries/raw/master/$(CONFIGURED_ARCH)/sai-plugin/$(BRANCH)/
-- 
2.25.1

