From 933ebe911370cc0755634d77cb838d83cd81dd12 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Tue, 25 Nov 2025 15:28:43 +0200
Subject: [PATCH 1/1] swss: ACL fix remove mem-override

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 orchagent/aclorch.cpp | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/orchagent/aclorch.cpp b/orchagent/aclorch.cpp
index 110cd223..b2b0ce63 100755
--- a/orchagent/aclorch.cpp
+++ b/orchagent/aclorch.cpp
@@ -3399,19 +3399,23 @@ bool AclRange::remove(sai_acl_range_type_t type, int min, int max)
 bool AclRange::remove(sai_object_id_t *oids, int oidsCnt)
 {
     SWSS_LOG_ENTER();
+    int removed = 0;
 
-    for (int oidIdx = 0; oidIdx < oidsCnt; oidsCnt++)
+    for (int oidIdx = 0; oidIdx < oidsCnt; oidIdx++)
     {
         for (auto it : m_ranges)
         {
-            if (it.second->m_oid == oids[oidsCnt])
+            if (it.second->m_oid == oids[oidIdx])
             {
-                return it.second->remove();
+                if (it.second->remove())
+                    removed++;
             }
         }
     }
 
-    return false;
+    SWSS_LOG_INFO("ACL remove OK=%d (%d vs %d)", (removed == oidsCnt), removed, oidsCnt);
+
+    return (removed == oidsCnt);
 }
 
 bool AclRange::remove()
-- 
2.34.1

