From d9d3662c61e15eb2c3fc85abfaf7ab93632bbd40 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Mon, 6 Oct 2025 00:06:15 +0300
Subject: [PATCH 1/1] sonic-platform-marvell rules rd98dx45xx rm

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 debian/rules | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/debian/rules b/debian/rules
index 3450dd4..21a5bbc 100755
--- a/debian/rules
+++ b/debian/rules
@@ -12,7 +12,7 @@ endif
 KERNEL_SRC :=  /lib/modules/$(KVERSION)
 INSTALL_MOD_DIR := kernel/extra/
 MOD_SRC_DIR:= $(shell pwd)
-MODULE_DIRS:= rd98dx35xx_cn9131 rd98dx35xx rd98dx35xx-x86 db98cx8580-32cd db98cx8514-10cc db98cx8540-16cd db98cx8522-10cc rd98dx45xx_cn9131
+MODULE_DIRS:= rd98dx35xx_cn9131 rd98dx35xx rd98dx35xx-x86 db98cx8580-32cd db98cx8514-10cc db98cx8540-16cd db98cx8522-10cc
 UTILS_DIR := utils
 SERVICE_DIR := service
 MRVL_MODULE_DIR:= mrvl-modules
@@ -43,8 +43,6 @@ build:
 		make clean; \
 		if [ $$mod = "rd98dx35xx_cn9131" ]; then \
 			make modules -C $(KERNEL_SRC)/build M=`pwd` CONFIG_KM_MVPCI=y CONFIG_KM_MVINT=y || exit 1; \
-		elif [ $$mod = "rd98dx45xx_cn9131" ]; then \
-			make modules -C $(KERNEL_SRC)/build M=`pwd` CONFIG_KM_MVPCI=y CONFIG_KM_MVINT=y || exit 1; \
         elif [ $$mod = "rd98dx35xx-x86" ];then \
 			make modules -C $(KERNEL_SRC)/build M=`pwd` CONFIG_KM_MVPCI=y CONFIG_KM_MVINT=y CONFIG_KM_MVDMA2=y|| exit 1; \
 		elif [ $$mod = "rd98dx35xx" ]; then \
@@ -79,8 +77,6 @@ binary-indep:
 	(for mod in $(MODULE_DIRS); do \
 		if [ $$mod = "rd98dx35xx_cn9131" ]; then \
 			pkg="rd98dx35xx-cn9131"; \
-		elif [ $$mod = "rd98dx45xx_cn9131" ]; then \
-			pkg="rd98dx45xx-cn9131"; \
 		else \
 			pkg=$$mod; \
 		fi; \
-- 
2.34.1

