From 29b9800c716dcb9a3e70e05d21521ee4f157c08f Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Thu, 18 Dec 2025 19:18:27 +0200
Subject: [PATCH 1/1] mrvl-pcie-ep-check.service - SMDEP ignore symlink

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 platform/marvell-prestera/mrvl-prestera.dep | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/platform/marvell-prestera/mrvl-prestera.dep b/platform/marvell-prestera/mrvl-prestera.dep
index 3ee32c3cb..f307900b7 100644
--- a/platform/marvell-prestera/mrvl-prestera.dep
+++ b/platform/marvell-prestera/mrvl-prestera.dep
@@ -2,6 +2,10 @@ MPATH       := $($(MRVL_PRESTERA_DEB)_SRC_PATH)
 DEP_FILES   := $(SONIC_COMMON_FILES_LIST) $(PLATFORM_PATH)/mrvl-prestera.mk $(PLATFORM_PATH)/mrvl-prestera.dep
 DEP_FILES   += $(SONIC_COMMON_BASE_FILES_LIST)
 SMDEP_FILES := $(addprefix $(MPATH)/,$(shell cd $(MPATH) && git ls-files))
+# Filter out the run-time symlink that is not in build-source tree
+SLINKS := $(shell find $(MPATH)/platform/*/common/etc/systemd -type l -exec echo {} \; | grep -Ev ' ')
+SLINKS += $(shell find $(MPATH)/platform/common/etc/systemd -type l -exec echo {} \; | grep -Ev ' ')
+SMDEP_FILES := $(filter-out $(SLINKS),$(SMDEP_FILES))
 
 $(MRVL_PRESTERA_DEB)_CACHE_MODE  := GIT_CONTENT_SHA
 $(MRVL_PRESTERA_DEB)_DEP_FLAGS   := $(SONIC_COMMON_FLAGS_LIST)
-- 
2.34.1

