From 804567c09de9fdc446ebbe59e8f9f597641d3143 Mon Sep 17 00:00:00 2001
From: nipunjoy <njoy@marvell.com>
Date: Mon, 14 Oct 2024 10:03:04 +0000
Subject: [PATCH] fix ICCPD make option

Signed-off-by: nipunjoy <njoy@marvell.com>
---
 Makefile.work | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.work b/Makefile.work
index 6079906a1005..e6cddad289b9 100644
--- a/Makefile.work
+++ b/Makefile.work
@@ -536,6 +536,7 @@ SONIC_BUILD_INSTRUCTION :=  $(MAKE) \
                            INCLUDE_DHCP_RELAY=$(INCLUDE_DHCP_RELAY) \
                            INCLUDE_DHCP_SERVER=$(INCLUDE_DHCP_SERVER) \
                            INCLUDE_MACSEC=$(INCLUDE_MACSEC) \
+                           INCLUDE_ICCPD=$(INCLUDE_ICCPD) \
                            SONIC_INCLUDE_RESTAPI=$(INCLUDE_RESTAPI) \
                            SONIC_INCLUDE_MUX=$(INCLUDE_MUX) \
                            ENABLE_TRANSLIB_WRITE=$(ENABLE_TRANSLIB_WRITE) \
