From 4cddd223e7ce92c248d8154a2a95057300ac33a7 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Mon, 9 Mar 2026 19:21:27 +0200
Subject: [PATCH 1/1] possibility to enable INCLUDE_ICCPD over env

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 rules/config | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rules/config b/rules/config
index f49dbbde9..3e1ea2991 100644
--- a/rules/config
+++ b/rules/config
@@ -169,7 +169,7 @@ INCLUDE_SYSTEM_TELEMETRY = n
 INCLUDE_SYSTEM_OTEL = y
 
 # INCLUDE_ICCPD - build docker-iccpd for mclag support
-INCLUDE_ICCPD = n
+INCLUDE_ICCPD ?= n
 
 # INCLUDE_SFLOW - build docker-sflow for sFlow support
 INCLUDE_SFLOW = y
-- 
2.34.1

