From 2d561b25c3b416c4c1d945e8eef94853f667f410 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Mon, 6 Jul 2026 13:04:43 +0300
Subject: [PATCH] marvell-prestera: sai-1.18.1-1

Update marvell-prestera SAI version from 1.17.1-13 onto 1.18.1-1

Signed-off-by: Yan Markman <ymarkman@marvell.com>

diff --git a/platform/marvell-prestera/sai.mk b/platform/marvell-prestera/sai.mk
index 18f41ff13..454a17a14 100644
--- a/platform/marvell-prestera/sai.mk
+++ b/platform/marvell-prestera/sai.mk
@@ -2,11 +2,11 @@
 
 BRANCH = master
 ifeq ($(CONFIGURED_ARCH),arm64)
-MRVL_SAI_VERSION = 1.17.1-13
+MRVL_SAI_VERSION = 1.18.1-1
 else ifeq ($(CONFIGURED_ARCH),armhf)
-MRVL_SAI_VERSION = 1.17.1-13
+MRVL_SAI_VERSION = 1.18.1-1
 else
-MRVL_SAI_VERSION = 1.17.1-13
+MRVL_SAI_VERSION = 1.18.1-1
 endif
 
 MRVL_SAI_URL_PREFIX = https://github.com/Marvell-switching/sonic-marvell-binaries/raw/master/$(CONFIGURED_ARCH)/sai-plugin/$(BRANCH)/
-- 
2.34.1

