diff --git a/platform/marvell-arm64/one-image.mk b/platform/marvell-arm64/one-image.mk
index b77b0d0..68d8f51 100644
--- a/platform/marvell-arm64/one-image.mk
+++ b/platform/marvell-arm64/one-image.mk
@@ -7,6 +7,7 @@ $(SONIC_ONE_IMAGE)_INSTALLS += $(SYSTEMD_SONIC_GENERATOR)
 $(SONIC_ONE_IMAGE)_INSTALLS += $(LINUX_KERNEL_DTB)
 $(SONIC_ONE_IMAGE)_LAZY_INSTALLS += $(FALCON_DB98CX8540_16CD_PLATFORM)
 $(SONIC_ONE_IMAGE)_LAZY_INSTALLS += $(FALCON_DB98CX8580_32CD_PLATFORM)
+$(SONIC_ONE_IMAGE)_LAZY_INSTALLS += $(AC5X_RD98DX7312_PLATFORM)
 
 ifeq ($(INSTALL_DEBUG_TOOLS),y)
 $(SONIC_ONE_IMAGE)_DOCKERS += $(SONIC_INSTALL_DOCKER_DBG_IMAGES)
diff --git a/platform/marvell-arm64/platform-rd98dx7312.mk b/platform/marvell-arm64/platform-rd98dx7312.mk
index e69de29..bcabe87 100644
--- a/platform/marvell-arm64/platform-rd98dx7312.mk
+++ b/platform/marvell-arm64/platform-rd98dx7312.mk
@@ -0,0 +1,8 @@
+# AC5X Platform
+
+AC5X_VERSION=0.1
+AC5X_RD98DX7312_PLATFORM = sonic-platform-rd98dx7312_$(AC5X_VERSION)_$(CONFIGURED_ARCH).deb
+$(AC5X_RD98DX7312_PLATFORM)_SRC_PATH = $(PLATFORM_PATH)/sonic-platform-rd98dx7312
+$(AC5X_RD98DX7312_PLATFORM)_PLATFORM = arm64-marvell_rd98DX7312_32G16HVG6HLG-r0
+SONIC_DPKG_DEBS += $(AC5X_RD98DX7312_PLATFORM)
+
diff --git a/platform/marvell-arm64/rules.mk b/platform/marvell-arm64/rules.mk
index b44fb4f..cb862cc 100644
--- a/platform/marvell-arm64/rules.mk
+++ b/platform/marvell-arm64/rules.mk
@@ -8,6 +8,7 @@ include $(PLATFORM_PATH)/one-image.mk
 include $(PLATFORM_PATH)/linux-kernel-arm64.mk
 include $(PLATFORM_PATH)/platform-db98cx8540-16cd.mk
 include $(PLATFORM_PATH)/platform-db98cx8580-32cd.mk
+include $(PLATFORM_PATH)/platform-rd98dx7312.mk
 
 
 SONIC_ALL += $(SONIC_ONE_IMAGE) \
diff --git a/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/db98cx8540/scripts/db98cx8540-16cd-init.sh b/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/db98cx8540/scripts/db98cx8540-16cd-init.sh
index f65a87b..756daf2 100644
--- a/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/db98cx8540/scripts/db98cx8540-16cd-init.sh
+++ b/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/db98cx8540/scripts/db98cx8540-16cd-init.sh
@@ -21,6 +21,8 @@ load_kernel_drivers() {
 
 # Install kernel drivers required for i2c bus access
 load_kernel_drivers
+#entropy setting
+python /etc/entropy.py
 
 # LOGIC to enumerate SFP eeprom devices - send 0x50 to kernel i2c driver - initialize devices
 # the mux may be enumerated at number 4 or 5 so we check for the mux and skip if needed
diff --git a/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/db98cx8540/scripts/db98cx8540-16cd_plt_setup.sh b/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/db98cx8540/scripts/db98cx8540-16cd_plt_setup.sh
index e69de29..81fe7c1 100755
--- a/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/db98cx8540/scripts/db98cx8540-16cd_plt_setup.sh
+++ b/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/db98cx8540/scripts/db98cx8540-16cd_plt_setup.sh
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+fw_uboot_env_cfg()
+{
+    echo "Setting up U-Boot environment..."
+    MACH_FILE="/host/machine.conf"
+    PLATFORM=`sed -n 's/onie_platform=\(.*\)/\1/p' $MACH_FILE`
+
+    FW_ENV_DEFAULT='/dev/mtd1 0x0 0x10000 0x100000'
+
+    echo $FW_ENV_DEFAULT > /etc/fw_env.config
+}
+
+db98cx8540_profile()
+{
+    MAC_ADDR=`ip link show eth0 | grep ether | awk '{print $2}'`
+    find /usr/share/sonic/device/*db98cx* -name profile.ini | xargs sed -i "s/switchMacAddress=.*/switchMacAddress=$MAC_ADDR/g"
+    echo "db98cx8540: Updating switch mac address ${MAC_ADDR}"
+}
+
+update_modulelist()
+{
+    MODULE_FILE="/etc/modules-load.d/marvell.conf"
+
+    echo "# Module list to load during the boot" > $MODULE_FILE
+    echo "mvIntDrv" >> $MODULE_FILE
+}
+
+main()
+{
+    fw_uboot_env_cfg
+    db98cx8540_profile
+    update_modulelist
+}
+
+main $@
diff --git a/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/db98cx8540/scripts/entropy.py b/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/db98cx8540/scripts/entropy.py
index e69de29..338e2ad 100644
--- a/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/db98cx8540/scripts/entropy.py
+++ b/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/db98cx8540/scripts/entropy.py
@@ -0,0 +1,19 @@
+#!/usr/bin/python
+import fcntl, struct
+import time
+from os import path
+
+RNDADDENTROPY=0x40085203
+
+def avail():
+  with open("/proc/sys/kernel/random/entropy_avail", mode='r') as avail:
+      return int(avail.read())
+
+if path.exists("/proc/sys/kernel/random/entropy_avail"):
+    while 1:
+        while avail() < 2048:
+            with open('/dev/urandom', 'rb') as urnd, open("/dev/random", mode='wb') as rnd:
+                d = urnd.read(512)
+                t = struct.pack('ii', 4 * len(d), len(d)) + d
+                fcntl.ioctl(rnd, RNDADDENTROPY, t)
+        time.sleep(30)
diff --git a/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/debian/sonic-platform-db98cx8540-16cd-db98cx8540.install b/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/debian/sonic-platform-db98cx8540-16cd-db98cx8540.install
index ded191a..bd7e8cd 100644
--- a/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/debian/sonic-platform-db98cx8540-16cd-db98cx8540.install
+++ b/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/debian/sonic-platform-db98cx8540-16cd-db98cx8540.install
@@ -1,5 +1,6 @@
+db98cx8540/scripts/db98cx8540-16cd_plt_setup.sh usr/local/bin
+db98cx8540/scripts/entropy.py etc/
 db98cx8540/scripts/db98cx8540-16cd-init.sh  usr/local/bin
 db98cx8540/service/db98cx8540-16cd-init.service etc/systemd/system
 db98cx8540/sonic_platform-1.0-py3-none-any.whl usr/share/sonic/device/arm64-marvell_db98cx8540_16cd-r0
 db98cx8540/sonic_platform-1.0-py2-none-any.whl usr/share/sonic/device/arm64-marvell_db98cx8540_16cd-r0
-
diff --git a/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/debian/sonic-platform-db98cx8540-16cd-db98cx8540.postinst b/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/debian/sonic-platform-db98cx8540-16cd-db98cx8540.postinst
index 1ae6cdc..e99887b 100644
--- a/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/debian/sonic-platform-db98cx8540-16cd-db98cx8540.postinst
+++ b/platform/marvell-arm64/sonic-platform-db98cx8540-16cd/debian/sonic-platform-db98cx8540-16cd-db98cx8540.postinst
@@ -3,6 +3,7 @@
 #
 # see: dh_installdeb(1)
 
+sh /usr/local/bin/db98cx8540-16cd_plt_setup.sh
 chmod a+x /usr/local/bin/db98cx8540-16cd-init.sh
 systemctl enable db98cx8540-16cd-init.service
 systemctl start db98cx8540-16cd-init.service
diff --git a/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/db98cx8580/scripts/db98cx8580-32cd-init.sh b/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/db98cx8580/scripts/db98cx8580-32cd-init.sh
index e463f5f..dcc725b 100644
--- a/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/db98cx8580/scripts/db98cx8580-32cd-init.sh
+++ b/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/db98cx8580/scripts/db98cx8580-32cd-init.sh
@@ -21,6 +21,8 @@ load_kernel_drivers() {
 
 # Install kernel drivers required for i2c bus access
 load_kernel_drivers
+#entropy setting
+python /etc/entropy.py
 
 # LOGIC to enumerate SFP eeprom devices - send 0x50 to kernel i2c driver - initialize devices
 # the mux may be enumerated at number 4 or 5 so we check for the mux and skip if needed
diff --git a/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/db98cx8580/scripts/db98cx8580-32cd_plt_setup.sh b/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/db98cx8580/scripts/db98cx8580-32cd_plt_setup.sh
index e69de29..d06c8c0 100755
--- a/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/db98cx8580/scripts/db98cx8580-32cd_plt_setup.sh
+++ b/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/db98cx8580/scripts/db98cx8580-32cd_plt_setup.sh
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+fw_uboot_env_cfg()
+{
+    echo "Setting up U-Boot environment..."
+    MACH_FILE="/host/machine.conf"
+    PLATFORM=`sed -n 's/onie_platform=\(.*\)/\1/p' $MACH_FILE`
+
+    FW_ENV_DEFAULT='/dev/mtd1 0x0 0x10000 0x100000'
+
+    echo $FW_ENV_DEFAULT > /etc/fw_env.config
+}
+
+db98cx8580_profile()
+{
+    MAC_ADDR=`ip link show eth0 | grep ether | awk '{print $2}'`
+    find /usr/share/sonic/device/*db98cx* -name profile.ini | xargs sed -i "s/switchMacAddress=.*/switchMacAddress=$MAC_ADDR/g"
+    echo "db98cx8540: Updating switch mac address ${MAC_ADDR}"
+}
+
+update_modulelist()
+{
+    MODULE_FILE="/etc/modules-load.d/marvell.conf"
+
+    echo "# Module list to load during the boot" > $MODULE_FILE
+    echo "mvIntDrv" >> $MODULE_FILE
+}
+
+main()
+{
+    fw_uboot_env_cfg
+    db98cx8580_profile
+    update_modulelist
+}
+
+main $@
diff --git a/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/db98cx8580/scripts/entropy.py b/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/db98cx8580/scripts/entropy.py
index e69de29..338e2ad 100644
--- a/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/db98cx8580/scripts/entropy.py
+++ b/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/db98cx8580/scripts/entropy.py
@@ -0,0 +1,19 @@
+#!/usr/bin/python
+import fcntl, struct
+import time
+from os import path
+
+RNDADDENTROPY=0x40085203
+
+def avail():
+  with open("/proc/sys/kernel/random/entropy_avail", mode='r') as avail:
+      return int(avail.read())
+
+if path.exists("/proc/sys/kernel/random/entropy_avail"):
+    while 1:
+        while avail() < 2048:
+            with open('/dev/urandom', 'rb') as urnd, open("/dev/random", mode='wb') as rnd:
+                d = urnd.read(512)
+                t = struct.pack('ii', 4 * len(d), len(d)) + d
+                fcntl.ioctl(rnd, RNDADDENTROPY, t)
+        time.sleep(30)
diff --git a/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/debian/sonic-platform-db98cx8580-32cd-db98cx8580.install b/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/debian/sonic-platform-db98cx8580-32cd-db98cx8580.install
index d63dab3..22e6234 100644
--- a/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/debian/sonic-platform-db98cx8580-32cd-db98cx8580.install
+++ b/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/debian/sonic-platform-db98cx8580-32cd-db98cx8580.install
@@ -1,3 +1,5 @@
+db98cx8580/scripts/db98cx8580-32cd_plt_setup.sh usr/local/bin
+db98cx8580/scripts/entropy.py etc/
 db98cx8580/scripts/db98cx8580-32cd-init.sh  usr/local/bin
 db98cx8580/service/db98cx8580-32cd-init.service etc/systemd/system
 db98cx8580/sonic_platform-1.0-py3-none-any.whl usr/share/sonic/device/arm64-marvell_db98cx8580_32cd-r0
diff --git a/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/debian/sonic-platform-db98cx8580-32cd-db98cx8580.postinst b/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/debian/sonic-platform-db98cx8580-32cd-db98cx8580.postinst
index ebabe73..6d59864 100644
--- a/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/debian/sonic-platform-db98cx8580-32cd-db98cx8580.postinst
+++ b/platform/marvell-arm64/sonic-platform-db98cx8580-32cd/debian/sonic-platform-db98cx8580-32cd-db98cx8580.postinst
@@ -3,6 +3,7 @@
 #
 # see: dh_installdeb(1)
 
+sh /usr/local/bin/db98cx8580-32cd_plt_setup.sh
 chmod a+x /usr/local/bin/db98cx8580-32cd-init.sh
 systemctl enable db98cx8580-32cd-init.service
 systemctl start db98cx8580-32cd-init.service
diff --git a/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/changelog b/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/changelog
index e69de29..7d0e185 100755
--- a/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/changelog
+++ b/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/changelog
@@ -0,0 +1,5 @@
+sonic-platform-rd98dx7312 (0.1) unstable; urgency=low
+
+  * Add support for rd98dx7312.
+
+ -- Marvell <pnaregundi@marvell.com>  Mon, 30 Nov 2021 09:35:58 +0800
diff --git a/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/compat b/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/compat
index e69de29..ec63514 100644
--- a/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/compat
+++ b/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/compat
@@ -0,0 +1 @@
+9
diff --git a/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/control b/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/control
index e69de29..dc4f442 100755
--- a/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/control
+++ b/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/control
@@ -0,0 +1,13 @@
+Source: sonic-platform-rd98dx7312
+Section: unknown
+Priority: optional
+Maintainer: Marvell <pnaregundi@marvell.com>
+Build-Depends: debhelper (>=9)
+Standards-Version: 3.9.6
+Homepage: <insert the upstream URL, if relevant>
+
+Package: sonic-platform-rd98dx7312
+Architecture: arm64
+Depends: ${misc:Depends}
+Description: <insert up to 60 chars description>
+ <insert long description, indented with spaces>
diff --git a/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/install b/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/install
index e69de29..346bcf7 100644
--- a/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/install
+++ b/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/install
@@ -0,0 +1,4 @@
+rd98dx7312/scripts/rd98dx7312_plt_setup.sh usr/local/bin
+rd98dx7312/scripts/entropy.py etc/
+rd98dx7312/scripts/rd98dx7312-init.sh usr/local/bin
+rd98dx7312/service/rd98dx7312-init.service etc/systemd/system
diff --git a/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/postinst b/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/postinst
index e69de29..587bb30 100644
--- a/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/postinst
+++ b/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/postinst
@@ -0,0 +1,44 @@
+#!/bin/sh
+# postinst script for sonic-platform-rd98dx7312
+#
+# see: dh_installdeb(1)
+
+set -e
+
+# summary of how this script can be called:
+#        * <postinst> `configure' <most-recently-configured-version>
+#        * <old-postinst> `abort-upgrade' <new version>
+#        * <conflictor's-postinst> `abort-remove' `in-favour' <package>
+#          <new-version>
+#        * <postinst> `abort-remove'
+#        * <deconfigured's-postinst> `abort-deconfigure' `in-favour'
+#          <failed-install-package> <version> `removing'
+#          <conflicting-package> <version>
+# for details, see https://www.debian.org/doc/debian-policy/ or
+# the debian-policy package
+
+
+case "$1" in
+    configure)
+	sh /usr/local/bin/rd98dx7312_plt_setup.sh
+	chmod a+x /usr/local/bin/rd98dx7312-init.sh
+	systemctl enable rd98dx7312-init.service
+	systemctl start rd98dx7312-init.service
+
+    ;;
+
+    abort-upgrade|abort-remove|abort-deconfigure)
+    ;;
+
+    *)
+        echo "postinst called with unknown argument \`$1'" >&2
+        exit 1
+    ;;
+esac
+
+# dh_installdeb will replace this with shell code automatically
+# generated by other debhelper scripts.
+
+#DEBHELPER#
+
+exit 0
diff --git a/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/rules b/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/rules
index e69de29..2f54c47 100755
--- a/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/rules
+++ b/platform/marvell-arm64/sonic-platform-rd98dx7312/debian/rules
@@ -0,0 +1,57 @@
+#!/usr/bin/make -f
+# See debhelper(7) (uncomment to enable)
+# output every command that modifies files on the build system.
+#export DH_VERBOSE = 1
+
+include /usr/share/dpkg/pkg-info.mk
+#--------------------------------------------------------
+
+PACKAGE_PRE_NAME := sonic-platform-rd98dx7312
+MOD_SRC_DIR:= $(shell pwd)
+MODULE_DIRS:= rd98dx7312
+SERVICE_DIR := service
+
+%:
+       dh $@ --with systemd
+
+clean:
+       dh_testdir
+       dh_testroot
+       dh_clean
+
+build:
+       # Nothing to do
+
+binary: binary-arch binary-indep
+       # Nothing to do
+
+binary-arch:
+       # Nothing to do
+
+binary-indep:
+       dh_testdir
+       dh_installdirs
+
+       # Custom package commands
+       (for mod in $(MODULE_DIRS); do \
+               dh_installdirs -p$(PACKAGE_PRE_NAME) /usr/local/bin; \
+       done)
+
+       # Resuming debhelper scripts
+       dh_testroot
+       dh_install
+       dh_installchangelogs
+       dh_installdocs
+       dh_systemd_enable
+       dh_installinit
+       dh_systemd_start
+       dh_link
+       dh_fixperms
+       dh_compress
+       dh_strip
+       dh_installdeb
+       dh_gencontrol
+       dh_md5sums
+       dh_builddeb
+
+.PHONY: build binary binary-arch binary-indep clean
diff --git a/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/scripts/entropy.py b/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/scripts/entropy.py
index e69de29..338e2ad 100644
--- a/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/scripts/entropy.py
+++ b/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/scripts/entropy.py
@@ -0,0 +1,19 @@
+#!/usr/bin/python
+import fcntl, struct
+import time
+from os import path
+
+RNDADDENTROPY=0x40085203
+
+def avail():
+  with open("/proc/sys/kernel/random/entropy_avail", mode='r') as avail:
+      return int(avail.read())
+
+if path.exists("/proc/sys/kernel/random/entropy_avail"):
+    while 1:
+        while avail() < 2048:
+            with open('/dev/urandom', 'rb') as urnd, open("/dev/random", mode='wb') as rnd:
+                d = urnd.read(512)
+                t = struct.pack('ii', 4 * len(d), len(d)) + d
+                fcntl.ioctl(rnd, RNDADDENTROPY, t)
+        time.sleep(30)
diff --git a/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/scripts/rd98dx7312-init.sh b/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/scripts/rd98dx7312-init.sh
index e69de29..d2229ed 100644
--- a/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/scripts/rd98dx7312-init.sh
+++ b/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/scripts/rd98dx7312-init.sh
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Platform init script
+
+# Load required kernel-mode drivers
+load_kernel_drivers() {
+    # Remove modules loaded during Linux init
+    # FIX-ME: This will be removed in the future when Linux init no longer loads these
+    rmmod i2c_dev
+    rmmod i2c_mv64xxx
+
+    # Carefully control the load order here to ensure consistent i2c bus numbering
+    modprobe i2c_mv64xxx
+    modprobe i2c_dev
+    modprobe eeprom
+    modprobe mvMbusDrv
+}
+
+# - Main entry
+
+# Install kernel drivers required for i2c bus access
+load_kernel_drivers
+#entropy setting
+python /etc/entropy.py
+
+exit 0
diff --git a/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/scripts/rd98dx7312_plt_setup.sh b/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/scripts/rd98dx7312_plt_setup.sh
index e69de29..26344bd 100755
--- a/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/scripts/rd98dx7312_plt_setup.sh
+++ b/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/scripts/rd98dx7312_plt_setup.sh
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+fw_uboot_env_cfg()
+{
+    echo "Setting up U-Boot environment..."
+    MACH_FILE="/host/machine.conf"
+    PLATFORM=`sed -n 's/onie_platform=\(.*\)/\1/p' $MACH_FILE`
+
+    FW_ENV_DEFAULT='/dev/mtd0 0x400000 0x10000 0x10000'
+
+    echo $FW_ENV_DEFAULT > /etc/fw_env.config
+}
+
+rd98DX7312_profile()
+{
+    MAC_ADDR=$(fw_printenv -n ethaddr)
+    find /usr/share/sonic/device/*rd98DX7312* -name profile.ini | xargs sed -i "s/switchMacAddress=.*/switchMacAddress=$MAC_ADDR/g"
+    echo "rd98DX7312: Updating switch mac address ${MAC_ADDR}"
+}
+
+update_modulelist()
+{
+    MODULE_FILE="/etc/modules-load.d/marvell.conf"
+
+    echo "# Module list to load during the boot" > $MODULE_FILE
+    echo "mvIntDrv" >> $MODULE_FILE
+}
+
+main()
+{
+    fw_uboot_env_cfg
+    rd98DX7312_profile
+    update_modulelist
+}
+
+main $@
diff --git a/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/service/rd98dx7312-init.service b/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/service/rd98dx7312-init.service
index e69de29..a863f27 100644
--- a/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/service/rd98dx7312-init.service
+++ b/platform/marvell-arm64/sonic-platform-rd98dx7312/rd98dx7312/service/rd98dx7312-init.service
@@ -0,0 +1,14 @@
+[Unit]
+Description=Platform rd98DX7312 Service
+Before=pmon.service
+After=sysinit.target
+DefaultDependencies=no
+
+[Service]
+ExecStart=/usr/local/bin/rd98dx7312-init.sh
+KillSignal=SIGKILL
+SuccessExitStatus=SIGKILL
+#StandardOutput=tty
+
+[Install]
+WantedBy=multi-user.target
