From a472c26a850b42bc1edf89d6838f4f6c2c2adb4a Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Thu, 16 Oct 2025 15:43:36 +0300
Subject: [PATCH 1/1] debian-rules fix install services

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 debian/rules | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/debian/rules b/debian/rules
index fcf4dee..134954b 100755
--- a/debian/rules
+++ b/debian/rules
@@ -72,7 +72,6 @@ override_dh_auto_build:
 		cd $(MOD_SRC_DIR); \
 	done)
 
-
 override_dh_auto_test:
 	# No tests to run
 
@@ -90,6 +89,8 @@ override_dh_auto_install:
 		fi; \
 		dh_installdirs -psonic-platform-$${pkg} /$(KERNEL_SRC)/$(INSTALL_MOD_DIR); \
 		dh_installdirs -psonic-platform-$${pkg} /usr/local/bin; \
+		dh_installdirs -psonic-platform-$${pkg} /lib/systemd/system; \
+		cp $(MOD_SRC_DIR)/$${mod}/$(SERVICE_DIR)/*.service debian/sonic-platform-$${pkg}/lib/systemd/system/; \
 		cp $(MOD_SRC_DIR)/$${mod}/$(UTILS_DIR)/* debian/sonic-platform-$${pkg}/usr/local/bin/; \
 		cp $(MOD_SRC_DIR)/$${mod}/$(MRVL_MODULE_DIR)/$(PRESTERA_MODULE_SRC)/mvcpss.ko debian/sonic-platform-$${pkg}/$(KERNEL_SRC)/$(INSTALL_MOD_DIR); \
         if [ $$mod = "db98cx8580-32cd" ] || \
-- 
2.34.1

