From 4b40c43555a619c3b61be7e8d022f96065ed7ffc Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Fri, 17 Oct 2025 10:30:25 +0300
Subject: [PATCH 1/1] platform submod [marvell] mrvl-prestera for trixie

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 platform/marvell-prestera/mrvl-prestera | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/platform/marvell-prestera/mrvl-prestera b/platform/marvell-prestera/mrvl-prestera
index 88f0532a0..1a70c9277 160000
--- a/platform/marvell-prestera/mrvl-prestera
+++ b/platform/marvell-prestera/mrvl-prestera
@@ -1 +1 @@
-Subproject commit 88f0532a0e6c33d399dc69096da463ce4fcfa8d4
+Subproject commit 1a70c9277aab706e85c8d0cff0b5fe9d2e37c084
-- 
2.34.1

