From 9836b0e81d525336952263708d5887af8f99ddc6 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Wed, 12 Nov 2025 14:48:16 +0200
Subject: [PATCH 1/1] [submod] mrvl-prestera KO update trixie

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 platform/marvell-prestera/mrvl-prestera | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/platform/marvell-prestera/mrvl-prestera b/platform/marvell-prestera/mrvl-prestera
index 1a70c9277..e3b309086 160000
--- a/platform/marvell-prestera/mrvl-prestera
+++ b/platform/marvell-prestera/mrvl-prestera
@@ -1 +1 @@
-Subproject commit 1a70c9277aab706e85c8d0cff0b5fe9d2e37c084
+Subproject commit e3b3090860dadcaef73e3fa41a1f1ba086c254b6
-- 
2.34.1

