From 09256d5098a23d007f534195a87770694bff7812 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Fri, 21 Nov 2025 23:36:19 +0200
Subject: [PATCH 1/1] orchagent/portsorch: fix Query ingress priority groups
 lists

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 orchagent/portsorch.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/orchagent/portsorch.cpp b/orchagent/portsorch.cpp
index 7f3c29f3..89f9a37e 100644
--- a/orchagent/portsorch.cpp
+++ b/orchagent/portsorch.cpp
@@ -6469,6 +6469,11 @@ void PortsOrch::initializePriorityGroupsBulk(std::vector<Port>& ports)
             const auto& port = ports[idx];
             const auto status = bulker.statuses[idx];
 
+            if (port.m_priority_group_ids.size() == 0)
+            {
+                continue;
+            }
+
             if (status != SAI_STATUS_SUCCESS)
             {
                 SWSS_LOG_ERROR("Fail to get priority group list for port %s rv:%d", port.m_alias.c_str(), status);
-- 
2.34.1

