From 805c984b26d3bbd88d303b556e087942cf653378 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Sun, 9 Nov 2025 21:47:46 +0200
Subject: [PATCH 1/1] [sonic-swss] marvell Nokia-armhf ignore PriorityGroups
 failure

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 orchagent/portsorch.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/orchagent/portsorch.cpp b/orchagent/portsorch.cpp
index 7f3c29f3..12b37b2f 100644
--- a/orchagent/portsorch.cpp
+++ b/orchagent/portsorch.cpp
@@ -6471,11 +6471,11 @@ void PortsOrch::initializePriorityGroupsBulk(std::vector<Port>& ports)
 
             if (status != SAI_STATUS_SUCCESS)
             {
-                SWSS_LOG_ERROR("Fail to get priority group list for port %s rv:%d", port.m_alias.c_str(), status);
-                handleSaiGetStatus(SAI_API_PORT, status);
-                throw runtime_error("PortsOrch initialization failure.");
+                SWSS_LOG_ERROR("Fail to get priority group list for port %s rv:%d. IGNORED", port.m_alias.c_str(), status);
+                //handleSaiGetStatus(SAI_API_PORT, status);
+                //throw runtime_error("PortsOrch initialization failure.");
             }
-
+            else
             SWSS_LOG_INFO("Get priority groups for port %s", port.m_alias.c_str());
         }
     }
-- 
2.34.1

