From 9891e6664e457c7aa6e83148e2327e993e847ccb Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Mon, 29 Sep 2025 14:47:08 +0300
Subject: [PATCH 1/1] sonic-platform-wistron: rules for trixie

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 .../sonic-platform-wistron/debian/rules       | 66 +++++++++----------
 1 file changed, 30 insertions(+), 36 deletions(-)

diff --git a/platform/marvell-prestera/sonic-platform-wistron/debian/rules b/platform/marvell-prestera/sonic-platform-wistron/debian/rules
index 328196af1..fbcb56b25 100755
--- a/platform/marvell-prestera/sonic-platform-wistron/debian/rules
+++ b/platform/marvell-prestera/sonic-platform-wistron/debian/rules
@@ -19,38 +19,51 @@ SERVICE_DIR := service
 PRESTERA_MODULE_SRC := mrvl-prestera/drivers/generic/cpssEnabler/linuxNoKernelModule/drivers
 
 %:
-	dh $@ --with systemd,python3 --buildsystem=pybuild
+	dh $@ --with python3 --buildsystem=pybuild
 
-clean:
-	dh_testdir
-	dh_testroot
+override_dh_auto_clean:
+	(for mod in $(MODULE_DIRS); do \
+		$(MAKE) clean -C $(KERNEL_SRC)/build M=$(MOD_SRC_DIR)/$${mod}/modules; \
+		if [ -f $(MOD_SRC_DIR)/$${mod}/setup.py ]; then \
+			PYBUILD_NAME=$${mod} pybuild --clean -d $${mod}; \
+		fi; \
+		cd $(MOD_SRC_DIR)/$${mod}; \
+		rm -rf $(MRVL_MODULE_DIR)/; \
+		rm -rf *.egg-info/; \
+		rm -f *.whl; \
+		cd $(MOD_SRC_DIR)/; \
+	done)
 	dh_clean
 
-build:
+override_dh_auto_configure:
+	(for mod in $(MODULE_DIRS); do \
+		if [ -f $(MOD_SRC_DIR)/$${mod}/setup.py ]; then \
+			PYBUILD_NAME=$${mod} pybuild --configure -d $${mod}; \
+		fi; \
+	done)
+
+override_dh_auto_build:
 	(for mod in $(MODULE_DIRS); do \
 		make modules -C $(KERNEL_SRC)/build M=$(MOD_SRC_DIR)/$${mod}/modules; \
 		cd $(MOD_SRC_DIR)/../$(PRESTERA_MODULE_SRC)/; \
 		make clean; \
 		make modules -C $(KERNEL_SRC)/build M=`pwd` CONFIG_KM_MVPCI=y CONFIG_KM_MVINT=y || exit 1; \
-                cp *.ko $(MOD_SRC_DIR)/$${mod}/$(MODULE_DIR)/; \
-        cd $(MOD_SRC_DIR)/$${mod}; \
+		cp *.ko $(MOD_SRC_DIR)/$${mod}/$(MODULE_DIR)/; \
+		cd $(MOD_SRC_DIR)/$${mod}; \
+		python3 setup.py build; \
 		python3 setup.py bdist_wheel -d $(MOD_SRC_DIR)/$${mod}; \
-        echo "Finished making whl package for $$mod"; \
 		cd $(MOD_SRC_DIR); \
 	done)
 
-binary: binary-arch binary-indep
-	# Nothing to do
 
-binary-arch:
-	# Nothing to do
+override_dh_auto_test:
+	# No tests to run
 
-binary-indep:
-	dh_testdir
-	dh_installdirs
+override_dh_usrlocal:
+	# debian/sonic-platform-<>-<>/usr/local/bin/* are already done
 
-	# Custom package commands
-	(for mod in $(MODULE_DIRS); do \
+override_dh_auto_install:
+	(set -e; for mod in $(MODULE_DIRS); do \
 		if [ $$mod = "es1227_54ts" ]; then \
 			pkg="es1227-54ts"; \
 		elif [ $$mod = "es2227_54ts" ]; then \
@@ -66,22 +79,3 @@ binary-indep:
 		cp $(MOD_SRC_DIR)/$${mod}/$(SERVICE_DIR)/*.service debian/sonic-platform-$${pkg}/lib/systemd/system/; \
 		#python3 $${mod}/setup.py install --root=$(MOD_SRC_DIR)/debian/sonic-platform-$${pkg} --install-layout=deb; \
 	done)
-
-	# Resuming debhelper scripts
-	dh_testroot
-	dh_install
-	dh_installchangelogs
-	dh_installdocs
-	dh_systemd_enable
-	dh_installinit
-	dh_systemd_start
-	dh_link
-	dh_fixperms
-	dh_compress
-	dh_strip
-	dh_installdeb
-	dh_gencontrol
-	dh_md5sums
-	dh_builddeb
-
-.PHONY: build binary binary-arch binary-indep clean
-- 
2.34.1

